modules/class-simplechart-wp-cli.php

```diff
@@ -0,0 +1,246 @@
+	 * @return array Post object transformed for our data loader
+	 */
+	private function _transform_from_wpcom_api( $post_object ) {
+		// basic post info
+		$post_data = array(
+			'post_name' => $post_object['slug'],
+			'post_title' => $post_object['title'],
+			'post_status' => $post_object['status'],
+			'post_type' => $this->_post_type,
+			'post_date_gmt' => date( 'Y-m-d H:i:s', strtotime( $post_object['date'] ) ),
+		);
+
+		// author ID if we have one
+		if ( $this->_author ) {
+			$post_data['post_author'] = $this->_author;
+		}
+
+		// setup post metadata
+		$post_meta = array(
+			'_src_guid' => $post_data['guid'],
+		);
+		if ( ! empty( $post_object['metadata'] ) && is_array( $post_object['metadata'] ) ) {
+			foreach ( $post_object['metadata']  as $meta ) {
+				if ( in_array( $meta['key'], $this->_meta_keys, true ) ) {
+					$post_meta[ $meta['key'] ] = $meta['value'];
+				}
+			}
+		}
+
+		// featured image URL
+		$featured_image = ! empty( $post_object['featured_image'] ) ? $post_object['featured_image'] : null;
+
+		return array(
+			'post_data' => $post_data,
+			'post_meta' => $post_meta,
+			'featured_image' => $featured_image,
+		);
+	}
+
+	/**
+	 * create new post for the chart, using already transformed data
+	 * @var array $data Post data for new chart
+	 *			array 'post_data' Arguments to pass to wp_insert_post()
+	 *			array 'post_meta' Key-value pairs to set as post meta
+	 *			array 'featured_image' Optional array of url, width, height of image to import
+	 * @return bool Success or failure
+	 */
+	private function _load_post( $data ) {
+		if ( empty( $data['post_data'] ) ) {
+			WP_CLI::warning( __( 'Cannot insert post with no post_data', 'simplechart' ) );
+			return false;
+		}
+
+		// create the post
+		$post_id = wp_insert_post( $data['post_data'] );
+		if ( 0 === $post_id ) {
+			WP_CLI::warning( __( 'Failed to insert post', 'simplechart' ) );
+			return false;
+		}
+
+		// add the post metadata
+		foreach ( $data['post_meta'] as $key => $value ) {
+			update_post_meta( $post_id, $key, $value );
+		}
+
+		// attach the featured image
+		if ( $data['featured_image'] ) {
+			$this->_attach_featured_image_to_post( $data['featured_image'], $post_id, $data['post_data']['post_title'] );
+		}
+
+		return true;
+	}
+
+	/**
+	 * sideload featured image from URL and set as post thumbnail
+	 * @var array $image Image URL
+	 * @var int $post_id ID of Simplechart post
+	 * @var string $desc Description for image
+	 */
+	private function _attach_featured_image_to_post( $image, $post_id, $desc ) {
+		$tmp = download_url( $image );
+		if ( is_wp_error( $tmp ) ) {
+			WP_CLI::warning( sprintf( __( 'Error downloading %s', 'simplechart' ), esc_url( $image ) ) );
+			if ( file_exists( $tmp ) ) {
+				@unlink( $tmp );
+			}
+			return;
+		}
+
+		$file_array = array(
+			'tmp_name' => $tmp,
+			'name' => basename( $image ),
+		);
+		$media_id = media_handle_sideload( $file_array, $post_id, $desc );
```

Comment:
Generally speaking, file system operations are prohibited on VIP. They say this will be ok but we'll see...